### DISCUSSION TITLE
How can I ping the database?

### DISCUSSION THREAD

**Discussion by PAdventures** [supabase/supabase#21908] [Questions] [answered]
Is there a way to ping the database?

I'm using Drizzle and I would like to include a feature in my app that will give info about the database connection ping, is there any way to do this?

**Comment by Nunyaklah** [comment]
Are you trying to get some specific information from the ping? I don't fully understand your use case but if you are trying to ping the database just to verify that it is reachable and responsive then you can execute a simple query against your database, such as selecting a small amount of data from a table.

**Reply by PAdventures** [reply]
So if you've worked with a discord bot, for example, you can access the WebSocket heartbeat as well as a latency roundtrip. This is what I would hope to get from Supabase. So how long it would take for my app to send and receive a response from my Supabase database. If that's not possible, then the average speed in the API reports is good enough.

**Reply by Nunyaklah** [reply]
Yeah this makes sense. This might also be useful :

```
  import supabase from "../config/supabaseClient";

  const YourComponent = () => {
       console.log(supabase) // you get details of your supabase connection here
       
       ....
   }
```

When you inspect your console you would get relevant information on your supabase client connection, for example `realtime.heartbeatIntervalMs` can be useful to you. There is a bunch of data in there that might be useful to your usecase.

Inside `supabaseClient.js` i just have the normal configuration guide supabase provides in its documentation.

**Comment by PAdventures** [comment, marked as answer]
I found the solution to get the latency  roundtrip for the DB connection
```js
const db = drizzle(...)
const beforeQuery = Date.now()
await db.execute(sql`select 1`)
const afterQuery = Date.now()
const latency = afterQuery - beforeQuery
```
